sdk/communication/azure-communication-rooms/azure/communication/rooms/_models/_models.py

```diff
@@ -3,52 +3,128 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------
-from typing import Any, Union
-from .._generated.models import (
-    ParticipantRole
-)
+from typing import Any, Union, overload, Mapping, Optional
+from enum import Enum
+from datetime import datetime
+
+from azure.core import CaseInsensitiveEnumMeta
+
+
 from .._shared.models import (
     CommunicationIdentifier,
     identifier_from_raw_id
 )
 
-class RoomParticipant():
+
+class ParticipantRole(str, Enum, metaclass=CaseInsensitiveEnumMeta):
+    """The role of a room participant. The default value is Attendee."""
+
+    PRESENTER = "Presenter"
+    ATTENDEE = "Attendee"
+    CONSUMER = "Consumer"
+
+
+class RoomParticipant:
     """A participant of the room.
 
     All required parameters must be populated in order to send to Azure.
 
     :ivar communication_identifier: Communication Identifier.
-    :vartype communication_identifier:~azure.communication.rooms._shared.models.CommunicationIdentifier
+    :vartype communication_identifier: ~azure.communication.rooms.CommunicationIdentifier
     :ivar role: Role Name.
     :vartype role: Union[str, ParticipantRole]
     """
-    def __init__(self,
-                *,
-                communication_identifier: CommunicationIdentifier,
-                role: Union[str, ParticipantRole] = ParticipantRole.ATTENDEE,
-                **kwargs: Any
+    @overload
+    def __init__(
+        self,
+        *,
+        communication_identifier: Union[str, CommunicationIdentifier],
```

Comment:
Why is the union of `string` and `CommunicationIdentifier`? Shouldn't this just be type `CommunicationIdentifier`?

Reply:
CommunicationIdentifier is not an enum though, it looks like just a normal class

Reply:
My intention here was that someone could pass in the `raw_id` in place of the CommunicationIdentifier. However I can remove this for now. We can always add an explicit parameter for this and make it mutually exclusive to the existing parameter later.